test/faktory_worker/job/job_integration_test.exs

```diff
@@ -0,0 +1,40 @@
+defmodule FaktoryWorker.JobIntegrationTest do
+  use ExUnit.Case
+
+  alias FaktoryWorker.Job
+
+  describe "perform_async/3" do
+    test "should send a new job to faktory" do
+      faktory_name = :"Test_#{random_string()}"
+      start_supervised!({FaktoryWorker, name: faktory_name, pool: [size: 1]})
+
+      queue_name = random_string()
+
+      opts = [
+        queue: queue_name,
+        faktory_name: faktory_name
+      ]
+
+      job = Job.create_job(SomeWorker, %{hey: "there!"}, opts)
+
+      Job.perform_async(job, opts)
+
+      assert_queue_size(queue_name, 1)
+
+      :ok = stop_supervised(faktory_name)
+    end
+  end
+
+  defp assert_queue_size(queue_name, expected_size) do
+    Process.sleep(20)
```

Comment:
Would fetch or an implemented worker allow us to avoid the sleep call?

Reply:
Yes having fetch will allow us to remove that since fetch does the waiting for us. This sleep is temporary and the lowest value I found before seeing the race condition.